src/gateway/run-server.js

```diff
@@ -75,9 +75,11 @@ export const runServer = async (
     app.set("trust proxy", true);
 
     // Start the gateway.
+    const runtimeMode = getRuntimeMode();
+    const logLevel = getLogLevel();
     const gatewayOptions: GatewayOptions = {
-        mode: getRuntimeMode(),
-        logger: getLogger(),
+        mode: runtimeMode,
+        logger: createLogger(runtimeMode, logLevel),
```

Comment:
This is part of the fix. We create the logger explicitly at the launch of the gateway.